packages/contracts/contracts/TellerV2.sol

```diff
@@ -794,12 +810,34 @@ contract TellerV2 is
         // Start with the original due date being 1 payment cycle since bid was accepted
         dueDate_ = bid.loanDetails.acceptedTimestamp + bid.terms.paymentCycle;
 
+        // Calculate due date if payment cycle is set to monthly
+        if (
+            bidPaymentCycleType[_bidId] ==
+            IMarketRegistry.PaymentCycleType.Monthly
+        ) {
+            dueDate_ = uint32(
+                BokkyPooBahsDateTimeLibrary.addMonths(
+                    bid.loanDetails.acceptedTimestamp,
+                    1
+                )
+            );
+        }
+
         // Calculate the cycle number the last repayment was made
-        uint32 delta = lastRepaidTimestamp(_bidId) -
+        uint32 lastRepaidTimestamp = lastRepaidTimestamp(_bidId);
+        uint32 delta = lastRepaidTimestamp -
             bid.loanDetails.acceptedTimestamp;
         if (delta > 0) {
             uint32 repaymentCycle = 1 + (delta / bid.terms.paymentCycle);
             dueDate_ += (repaymentCycle * bid.terms.paymentCycle);
+            if (
+                BokkyPooBahsDateTimeLibrary.isLeapYear(bid.loanDetails.acceptedTimestamp)
+            ) {
+                (uint year, uint month, uint day) = BokkyPooBahsDateTimeLibrary.timestampToDate(lastRepaidTimestamp);
+                if (month == 2 && day == 29) {
```

Comment:
This seems kind of hacky - i dont like it. Is there a cleaner way to fix these edge cases?  This also doesnt account for the 31st of certain months.

Reply:
if anything, maybe this could be changed to  

if (month ==2 && day == 29 || day ==31)

Reply:
the dateTime lib actually handles this for us by checking if the day is more than the days in the following month and sets it accordingly
added tests to confirm